### DISCUSSION TITLE
Add ride NavBar Link is not working

### DISCUSSION THREAD

**Discussion by liebeckm** [neuefische/questions#576] [Web] [answered]
Hey @ all,
I am currently working on my app for carpooling. I added a form component that adds the ride to a list with all existing rides (coded a hardcoded lib for that) on the page "add rides". Both, booked rides from other drivers as well as my offered rides are displayed on the "my rides" page.
I did a update on the form, so that the form disappears once the ride is submitted. When doing that, the navbar Link to "add rides"-page does not work any longer (there is no reload). To solve this, I added `onClick={() => {
          window.location.href = "/add-rides";
        }}` to the navbar link, but then the offered rides don't appear into the "Meine Fahrten" page (guess because of the reload). then, I tried to solve this with useLocalStorageState (switched useState everywhere in my code to useLocalStorageState), but now there is an error message "Can't resolve 'use-local-storage-state". Can someone suggest please how I can make the data stored correctly as well as make the navbar link work again?

Links: 
- [User Story](https://github.com/liebeckm/capstone-project/issues/7)
- [Form to add Rides, here I used local storage state](https://github.com/liebeckm/capstone-project/blob/add-ride/components/Form/RideAddForm.styled.js)
-[ Page where said Form is used](https://github.com/liebeckm/capstone-project/blob/add-ride/pages/add-rides/index.js)
- [Navbar with Link to "add-rides" page ](https://github.com/liebeckm/capstone-project/blob/add-ride/components/NavBar/NavBar.js)

Thank you!!

**Comment by PReichetanz** [comment]
Hi @liebeckm,
I currently don't have enough time to figure out the link problem, but:

At least regarding the last point "Can't resolve 'use-local-storage-state": this happens when you use a npm package which is not installed. Run `npm i use-local-storage-state` to fix this issue.

Update: you've installed `use-local-storage`, but this is a different package.

**Comment by PReichetanz** [comment, marked as answer]
Hi @liebeckm,

I've found a way to solve your problem.

## localStorage
You don't need to save all state in localStorage, the both states in `_app.js`, `rides` and `bookedRides` are enough. 

1. Install the correct package as mentioned above: `npm i use-local-storage-state`. Remove the other one with `npm uninstall use-local-storage`. 
2. Adapt the syntax for the `use-local-storage-state` hook like this:

```jsx
// _app.js
  const [rides, setRides] = useLocalStorageState('rides', {
    defaultValue: initalRides,
  });
  const [bookedRides, setBookedRides] = useLocalStorageState('bookedRides', {
    defaultValue: [],
  });
```

3. Reset all other instances to use the `useState` again. It's enough if the component where they are used knows about them.

## Adapt the NavBar.js
Remove the `onClick={() => {window.location.href = "/add-rides";}}` from the `NavBarLink` as it is not what you want here.

## Adapt `pages/add-rides/index.js`
The problem is that, after submitting the form, you are still on `/add-rides`. This is why the Link doesn't do anything.

What you can do is to set the states back to `false` to get back to the start:

1. I suggest to rename the second state to `isFormSubmitted`. 
2. First option: `useEffect` + TimeOut. This would reset the page after a 3 seconds:
```jsx
// pages/add-rides/index.js
useEffect(() => {
    if (isFormSubmitted) {
      setTimeout(() => {
        setIsRideAdded(false);
        setIsFormSubmitted(false);
      }, 3000);
    }
  }, [isFormSubmitted]);
```

3. Second option: add another button to the part which is rendered dependent on `isRideAdded`; on clicking the button, both states are set to false.
```jsx
// pages/add-rides/index.js
const handleReset = () => {
    setIsRideAdded(false);
    setIsFormSubmitted(false);
  };

return (
...
{isRideAdded && (
  <StyledPaymentInformation>
    Deine Fahrt wurde erfolgreich hinzugefügt!
    <button type="button" onClick={handleReset}>
      Weitere Fahrt hinzufügen
    </button>
  </StyledPaymentInformation>
)}
...
)
```
